pgcli/main.py

```diff
@@ -45,11 +45,7 @@
 
 from pgspecial.main import (PGSpecial, NO_QUERY, PAGER_OFF, PAGER_LONG_OUTPUT)
 import pgspecial as special
-try:
-    import keyring
-except:
-    # pep8 will be unhappy about this. But keyring is optional, and we better disable it if it won't load.
-    keyring = None
+
```

Comment:
IMHO keeping it as a global on top adds to readability (i.e. see at a glance which modules are, or will be, imported by the `main.py`). Even if it's just `keyring = None`.

But if you see some benefit of making it a member variable, then sure.

Reply:
I just feel it's a little weird to have a `keyring = None` as a global variable, then update and use it inside a class.

For the readability example, I am not sure, I would not trust those import statements at the beginning of a Python file. Because I have seen many code import modules inside functions.

I don't have a strong preference though. If you still prefer having a global `keyring = None`, I will change it in the next revision.

Reply:
> I just feel it's a little weird to have a keyring = None as a global variable, then update and use it inside a class.

This kind of pattern is often used with lazy loading. I would prefer that import on top. Even though it's a weak preference. :)

Reply:
Done, thanks.

BTW, I only have seen below pattern before.

```Python
try:
    import xxx
except ImportError:
    xxx = None
```

Reply:
The pattern I'm talking about usually looks something like this:

```
class MyClass:

    def __init__(self):
        self._expensive_resource = None

    @property
    def my_resource(self):
        if self._expensive_resource is None:
            self._expensive_resource = self.init_expensive_resource()
        return self._expensive_resource
```

and you would use it when you 1) only want to try loading something once, because it's slow or otherwise expensive and 2) only want to load it if it's asked for. Our case of `keyring` here, even if it's not exactly the same, is very similar.

Reply:
@j-bennet Thanks for sharing. It looks like a [singleton pattern](https://en.wikipedia.org/wiki/Singleton_pattern#Lazy_initialization), it also similar to what I implemented in previous revision (except I put `self.xxx = None` in the a normal method)

Reply:
Yep.